README.md

```diff
@@ -42,6 +42,43 @@ do that in the [yagna repository](https://github.com/golemfactory/yagna) and in
   API, the creation of a requestor agent should be straighforward and require minimal effort.
   You can use examples contained in this repository (blender and hashcat) as references.
 
+### Components
+
+There are a few components that are crucial for any requestor agent app:
+
+#### Engine
+
+The heart of the high-level API is the requestor's task runner engine (`yapapi.runner.Engine`).
+You tell it, among others, which package (VM image) will be used to run your tasks, 
+how much you'd like to pay and how many providers you'd like to involve in the execution.
+Finally, you feed it the worker script and a list of `Task` objects to execute on providers. 
+
+#### Worker script
+
+The `worker` will most likely be the very core of your requestor app. You need to define
+this function in your agent code and then you pass it to the runner Engine.
+
+It receives the a `WorkContext` (`yapapi.runner.ctx.WorkContext`) object that serves 
+as an interface between your script and the execution unit within the provider. 
+Using the work context, you define the steps that the provider needs to execute in order
+to complete the job you're giving them.
+
```

Comment:
The WorkContext allows for interaction with the container running on the provider. By the WorkContext methods, you can transfer files to / from the provider and execute commands on the container.

Reply:
@rad9k done